test/e2e/user_group_limit/user_group_limit_test.go

```diff
@@ -641,6 +644,81 @@ var _ = ginkgo.Describe("UserGroupLimit", func() {
 		checkUsage(groupTestType, url.QueryEscape(validGroup), sandboxQueue1, []*v1.Pod{usergroup1Sandbox1Pod1})
 	})
 
+	ginkgo.It("Verify_Queue_Name_With_Special_Characters", func() {
+		ginkgo.By("Create a queue with a name that includes all allowed special characters")
+		queueName := "root_test22-a_b_#_c_#_d_/_e@dom:ain"
+		yunikorn.UpdateCustomConfigMapWrapper(oldConfigMap, "", func(sc *configs.SchedulerConfig) error {
+			// remove placement rules so we can control queue
+			sc.Partitions[0].PlacementRules = nil
+			var err error
+			if err = common.AddQueue(sc, "default", "root", configs.QueueConfig{
+				Name:       queueName,
+				Resources:  configs.Resources{Guaranteed: map[string]string{"memory": fmt.Sprintf("%dM", 200)}},
+				Properties: map[string]string{"preemption.delay": "1s"},
+			}); err != nil {
+				return err
+			}
+			return nil
+		})
+		ginkgo.By("Fetch the queue information using the REST API")
+		allQueues, err := restClient.GetQueues("default")
+		gomega.Ω(err).NotTo(gomega.HaveOccurred())
+		ginkgo.By("Verify that the queue information is returned correctly")
+		// loop through allQueues.children to find the queue with the name we created
+		for _, queue := range allQueues.Children {
+			if queue.QueueName == "root."+queueName {
+				gomega.Ω(queue.QueueName).To(gomega.Equal("root." + queueName))
+			} else {
+				gomega.Ω(queue.QueueName).NotTo(gomega.Equal("root." + queueName))
+			}
+		}
```

Comment:
Since we started using direct API below, can we clean up the above code from line no. 664 to 674?

Reply:
Sure will remove this part of code.